test/Test/TypeRep/TypeRepMapProperty.hs

```diff
@@ -82,6 +87,38 @@ deleteInvariantSpec = it "invariantCheck (delete k b) == True" $ hedgehog $ do
     WrapTypeable (_ :: IntProxy n) <- forAll genTF
     assert $ invariantCheck (delete @n m)
 
+alterInsertSpec :: Property
+alterInsertSpec = it "insert proxy m === alter (const (Just proxy)) m" $ hedgehog $ do
+    m <- forAll genMap
+    WrapTypeable (proxy :: IntProxy n) <- forAll genTF
+    insert proxy m === alter (const (Just proxy)) m
+
+alterDeleteSpec :: Property
+alterDeleteSpec = it "delete proxy m === alter (const Nothing) m" $ hedgehog $ do
+    WrapTypeable (proxy :: IntProxy n) <- forAll genTF
+    m <- insert proxy <$> forAll genMap
+    delete @n @IntProxy m === alter @n @IntProxy (const Nothing) m
+
+alterAdjustSpec :: Property
+alterAdjustSpec = it "adjust f m == alter (fmap f) m" $ hedgehog $ do
+    m <- forAll genMap
+    WrapTypeable (_ :: IntProxy n) <- forAll genTF
+    randInt <- forAll (Gen.int Range.constantBounded)
+    let f (IntProxy p n) = IntProxy p (n * 10)
+    adjust @n @IntProxy f m === alter @n @IntProxy (fmap f) m
```

Comment:
Very nice properties! It looks, you need to use the same trick (described in my review for #96) to make the tests pass on older GHC versions